lib/Db/PimcoreExtensionsTrait.php

```diff
@@ -352,9 +352,34 @@ public function insertOrUpdate($table, array $data)
             $i++;
         }
 
+        // get the field name of the primary key
+        $schemaManager = $this->getSchemaManager();
+        $indexes = $schemaManager->listTableIndexes($table);
+        $fieldsPrimaryKeyRaw = [];
+        foreach ($indexes as $index) {
+            if ($index->isPrimary()) {
+                $fieldsPrimaryKeyRaw = (array)$index->getColumns();
+                break;
+            }
+        }
+
+        // format fieldnames for primary key to check in the following block
+        $fieldsPrimaryKey = [];
+        foreach ($fieldsPrimaryKeyRaw as $item) {
+            $fieldsPrimaryKey[] = $this->quoteIdentifier($item);
+        }
+
         // build the statement
         $set = [];
         foreach ($cols as $i => $col) {
+            if (
+                in_array($col, $fieldsPrimaryKey)
+                &&
+                $data[trim($col, $this->getDatabasePlatform()->getIdentifierQuoteCharacter())] === null
+            ) {
+                continue;
+            }
+
             $set[] = sprintf('%s = %s', $col, $vals[$i]);
```

Comment:
I think we can build the `$set` array directly within the `foreach` at line `350` above. Then we don't need to compare quoted identifiers and also make the code more readable.

Reply:
Ok, I will look into that within the next days. I cant finish that today and am away until friday.

Reply:
Ok, thanks! Maybe I have some time left in the meantime, then I can finish it for you. Anyway, thanks already a lot for pointing at the right place!